### PR TITLE
Adds corpse flowers

### PR DESCRIPTION
**Pull request by granpawalton**
## About The Pull Request

Corpse flowers are a type of plant that produces a very strong and terrible smell, hence the name. To represent this it produces a decent bit of miasma overtime 12-32 times as much as a corpse that scales with the stats of the seed. Its not as much as it sounds, I play tested, ran the numbers, and set constraints to make it hard yet not terrible to maximize production on.

- The harvest works the same as starthistle to reinforce the idea that it needs to be planted to produce miasma and that it is easier to make more of the plant
- The plant mutates from starthistle and starthistle no longer mutates into harebells as harebells natural spawn anyway
- The miasma production only starts when it is about to bloom
- Miasma production only works around normal pressure
- The speed goes off yield and the pressure constraints off potency
- The plant isn't the most potent for antag use due to the fact that you can just harvest it to stop it
- Cargo and Science point values have been lowered to a fourth of what they used to for miasma (which is still pretty high in my opinion)
- Miasma mining will be easier to setup but require more maintenance than monkey room strats
- Its worth noting since it inherits from starthistle it shares its same base long lifespan and high endurance

Multiple trays will be required for decent output so either 1. The botanists commit their department to it 2. atmos/sci takes over garden and is capped at 6-8 trays (what I see happening) or 3. The atmos/sci builds additional trays which is time consuming to the point where the output and setup time is close to monkey rooms

All of these use cases require maintaining the atmosphere constraints and plant needs so that would be additional time cost to consider

If anyone thinks the sprites are bad and wants to make more I wouldn't mind

## Why It's Good For The Game

Creates a new botany plant that goes in a different direction in overall design for variety

## changelog
:cl: Granpawalton 
add: Added a new plant called the corpse flower that produces miasma while it is harvestable
tweak: Harebells no longer mutate from starthistle
balance: miasma's science and cargo point values have been reduced by 75%
/:cl:

### CODE REVIEW

File: code/__DEFINES/is_helpers.dm [hunk 1]
```diff
@@ -209,6 +209,8 @@ GLOBAL_LIST_INIT(heavyfootmob, typecacheof(list(
 
 #define ismachinery(A) (istype(A, /obj/machinery))
 
+#define ishydroponics(A) (istype(A, /obj/machinery/hydroponics))
```
Comment (granpawalton) [inline code/__DEFINES/is_helpers.dm]: is this the best way of doing what I am trying to accomplish?

File: code/modules/hydroponics/grown/misc.dm [hunk 2]
```diff
@@ -28,6 +28,40 @@
...
+		return FALSE
+
+	var/turf/open/T = get_turf(parent)
+	if(T.return_air().return_pressure() > (ONE_ATMOSPHERE + potency/10 + 10)) // clouds can begin showing at around 50-60 potency in standard atmos
+		return TRUE
+
+	var/datum/gas_mixture/stank = new
+	ADD_GAS(/datum/gas/miasma, stank.gases)
+	stank.gases[/datum/gas/miasma][MOLES] = (yield + 6)*7*MIASMA_CORPSE_MOLES // this process is only being called about 1/7 as much as corpses
+	stank.temperature = T20C // without this the room would eventually freeze and miasma mining would be easier
```
Comment (granpawalton) [inline code/modules/hydroponics/grown/misc.dm]: The temperature thing should be a problem elsewhere in the code, can I fix it in this PR?

**Comment by necromanceranne** [comment]
This is a really cool addition. It encourages a lot of cool things for botany to provide for other departments; namely science, cargo and atmos (though why atmo would want to farm miasma is beyond me, they may want it for....reasons....)

**Comment by Swiftkor** [comment]
Great PR, makes atmosians and botany happy

**Comment by bgobandit** [comment]
see this is the kind of plant addition that makes sense and opens up gameplay

:+1:

**Comment by ATH1909** [comment]
Stank gun here we go

**Comment by Tlaltecuhtli** [comment]
do they grow on corpses?

**Comment by FantasticFwoosh** [comment]
I find this flower very interesting, but i only have a trivial complaint about the lack of nessecity to remove harebells as obviously all flowers are accessible as not-weeds now (the definition was collapsed, they are just less useful plants than obstructions, for bees benefit mostly) straight from vendor and through this provided mutation path. They could certainly share the mutation path as berries & citrus fruits do multiple possible results from a single plant without much effort unless this is a subject of your personal preference.
> here was my PR for reference #27214

With some further gene modifications this plant could produce O2 or similar gases i think, which interests me immensely since nothing similar else exists to use in hostile enviroments.

File: code/__DEFINES/is_helpers.dm (hunk 1 above)
Comment (Qustinnus) [reply code/__DEFINES/is_helpers.dm]: Helpers are for commonly used is types, I'm not sure if hydroponics trays checks are common enough to warrant this. Either way you should replace all typechecks for hydroponics trays with this helper if you really want to use a helper.

**Comment by granpawalton** [comment]
@Tlaltecuhtli no but i triple dog dare you to make one that does

@FantasticFwoosh I took it out so obtaining it would be more consistent for people trying to do atmos setups. I dont really understand why you want it there can you elaborate?

File: code/__DEFINES/is_helpers.dm (hunk 1 above)
Comment (granpawalton) [reply code/__DEFINES/is_helpers.dm]: Thanks ima remove it, I didnt know if it was cheaper to do or something for language reasons

**Comment by FantasticFwoosh** [comment]
I don't really have a lot to add that i haven't said already, its in-consequential to the PR in the end since hacked vendors also have harebell seeds so you wouldnt need to wait anyway, it was just a neat feature for continuity for the starthistle to establish it as a real plant and open up a option for botanists to apply a mutation on it which this PR replaces anyway. I rescind my complaint in reterospect.

File: code/__DEFINES/is_helpers.dm (hunk 1 above)
Comment (Qustinnus) [reply code/__DEFINES/is_helpers.dm]: for future reference, defines are interpreted by the compiler as what its defined at, so to the compiler  ishydroponics(A) just means istype(A, /obj/machinery/hydroponics) so theres no runtime difference.

**Comment by granpawalton** [comment]
After play testing it I want to double it from 6-16 times a corpses production to 12-32 times so it is more useful for casual botanists and then increasing the atmospheric constraints to keep it hard for people trying to maximize miasma production and giving a general mention of the constraints on the seed examine text/wiki

I dont think the normal corpse production rate is the issue due to it already encouraging body bags in medbay and monkey rooms still easily rivaling the corpse flower strategy

**Comment by Krysonism** [comment]
How does the rates compare to a traditional corpse farm setup with gibbed monkies? Kinda lame if the new method obsoletes the old method AND guts the price of miasma.

**Comment by Qustinnus** [comment]
monkey corpse farms sound dumb anyways

**Comment by granpawalton** [comment]
@Krysonism ill explain how monkey rooms work

1. get 150-300 monkey cubes by breaking into botany and putting some of his stack of 100 giant wheats into the biogenerator
2. set up a tile that can be gibbed by the xeno bio shield gen while also can keep monkeys from escaping
3. make 150 (max monkeys at one time) monkeys on that tile, gib then repeat until youre out of cubes
4. suck up and use miasma via portascrubbers or setup scrubber pipes

you gibb the monkeys because it is fastest way to make them produce miasma, temperature reqs dont apply, and the amount of monkeys(including corpses) allowed at one time is 150

Read the corpse flower test cases I put in description and assume each tray is 20 corpses (tier 1 gene manip).

Corpse flowers when compared to monkey rooms should be better for making production a little more detailed and front loading production so each strat should be selected to fit things like: which server you're on, what is your end goal, whats accessible to you, when in the round did you decide to start and so on

**Comment by Onule** [comment]
Yo, if/when this gets merged, I'll get some sprites up later on.

**Comment by ExcessiveUseOfCobblestone** [comment]
neat

**Pull request merged**